src/librados/IoCtxImpl.cc

```diff
@@ -780,6 +780,40 @@ int librados::IoCtxImpl::operate_read(const object_t& oid,
   return r;
 }
 
+int librados::IoCtxImpl::aio_operate_repair_read(const object_t& oid,
+				      ::ObjectOperation *o,
+				      AioCompletionImpl *c,
+				      uint64_t snapid,
+				      int flags, int32_t osdid, epoch_t e)
+{
+  if (!o->size())
+    return 0;
+
+  auto onack = new C_aio_Complete(c);
+
+  c->is_read = true;
+  c->io = this;
+
+  int op = o->ops[0].op.op;
+  ldout(client->cct, 10) << ceph_osd_op_name(op) << " oid=" << oid << " nspace=" << oloc.nspace << dendl;
+  // Prepend assert_interval op
```

Comment:
> instead of modifying the ObjectOperation, you can use the extra_ops arg for the Objecter prepare_* helpers.

it's a different use case. `prepare_read_op()` prepend the existing ops in `extra_ops` to the new read op allocated in it. but we want to prepend an `assert_interval` op to the existing read ops in `o`.

> adding an Objecter::prepare_repair_read() helper here would be good too

okay.